File: Backend/foodee_service/src/main/java/org/example/foodee_service/entity/Restaurant.java
package org.example.foodee_service.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.locationtech.jts.geom.Point;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long restaurantId;

    private String restaurantName;

    private String restaurantAddress;

    private String restaurantPhone;

    private BigDecimal restaurantAvgRating;

    private String restaurantAvgRatingText;

    private String restaurantPhotoUrl;

    private Double restaurantLatitude;

    private Double restaurantLongitude;

    private String restaurantDescription;

    private Integer restaurantStatus;

    private Boolean restaurantIsOpening;

    private Boolean restaurantIsAds;

    private Boolean restaurantIsBooking;

    private Boolean restaurantIsDelivery;

    private LocalDateTime restaurantCreatedAt;

    private String restaurantCreatedBy;

    private LocalDateTime restaurantUpdatedAt;

    private String restaurantUpdatedBy;

    @OneToOne
    private User owner;
    
    @ManyToMany
    private Set<Category> category;
//
//    @Column(columnDefinition = "Geometry")
//    private Point location;

    @PrePersist
    public void prePersist() {
        restaurantCreatedAt = LocalDateTime.now();
        restaurantCreatedBy = "SYSTEM";
    }

    @PreUpdate
    public void preUpdate() {
        restaurantUpdatedAt = LocalDateTime.now();
        restaurantUpdatedBy = "SYSTEM";
    }
}

File: Frontend/src/app/pages/Apps/Ecommerce/Products/interfaces/restaurant.ts
export class Restaurant {
  restaurantId : number;
  restaurantName : string;
  restaurantAddress : string;
  restaurantPhone : string;
  restaurantAvgRatingText : string;
  restaurantPhotoUrl : string;
  restaurantLatitude : string;
  restaurantLongitude : string;
  categoryName : [] | string[];
  restaurantDescription : string;
  restaurantStatus : string;
  restaurantIsOpening : boolean;
  restaurantIsAds : boolean;
  restaurantIsBooking : boolean;
  restaurantIsDelivery : boolean;
  userId : string;
  distance : number;

  constructor() {
    this.categoryName = [];
    this.restaurantId = 0
    this.restaurantName = ''
    this.restaurantAddress = ''
    this.restaurantPhone = ''
    this.restaurantAvgRatingText = ''
    this.restaurantPhotoUrl = ''
    this.restaurantLatitude = ''
    this.restaurantLongitude = ''
    this.restaurantDescription = ''
    this.restaurantStatus = ''
    this.restaurantIsOpening = false;
    this.restaurantIsAds = false;
    this.restaurantIsBooking = false;
    this.restaurantIsDelivery = false;
    this.userId = ''
    this.distance = 0
  }

}
